src/Component/Calculator.js

```diff
@@ -1,100 +1,76 @@
-      ) {
-        return;
-      }
-
-      setCalc(calc + value);
-      if (!operator.includes(value)) {
-        setResult((calc + value).toString());
-      }
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
     };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick = (e) => {
+    const data = calculate(this.state, e.target.textContent);
+    this.setState(data);
+  };
 
+  render() {
+    const { total, next, operation } = this.state;
+    const result = `${total || ''} ${operation || ''} ${next || ''}`;
     const createDigits = () => {
       const digits = [];
       for (let i = 9; i > 0; i -= 1) {
         digits.push(
-          <button
-            id="numbers"
-            type="submit"
-            onClick={() => updateCalc(i.toString())}
-            key={i}
-          >
+          <button id="numbers" type="submit" onClick={this.handleClick} key={i}>
             {i}
           </button>,
         );
       }
       return digits;
     };
 
-    const calculate = () => {
-      setCalc(calc.toString());
-    };
-
-    const deleteLast = () => {
-      if (calc === '') {
-        return;
-      }
-      const value = calc.slice(0, -1);
-      setCalc(value);
-    };
-
     return (
       <div className="App">
         <div className="calculator">
           <div className="display">
-            {result ? (
-              <span>
-                (
-                {result}
-                )
-              </span>
-            ) : ''}
-          &nbsp;
-            {calc || '0'}
+            <span>{result.trim() ? result : 0}</span>
           </div>
           <div className="operator-digits-container">
             <div className="digits">
-              <button type="submit" onClick={deleteLast}>
-                A/C
+              <button type="submit" onClick={this.handleClick}>
+                AC
               </button>
-              <button type="submit" onClick={deleteLast}>
+              <button type="submit" onClick={this.handleClick}>
                 +/-
               </button>
-              <button type="submit" onClick={deleteLast}>
+              <button type="submit" onClick={this.handleClick}>
                 %
               </button>
               {createDigits()}
-              <button id="zero" type="submit" onClick={() => updateCalc('0')}>
+              <button id="zero" type="submit" onClick={this.handleClick}>
                 0
               </button>
-              <button type="submit" onClick={() => updateCalc('.')}>
+              <button type="submit" onClick={this.handleClick}>
                 .
               </button>
             </div>
             <div className="operators">
-              <button type="submit" onClick={() => updateCalc('/')}>
+              <button type="submit" onClick={this.handleClick}>
                 /
               </button>
```

Comment:
- [ ] I am unable to perform a divide operation because you are not using the operation recognised in the `operate.js` file. You have `/` whereas the operate file has &#247;. Kindly fix this. You can use this HTML entity code to get the divide sign: **`&#247;`**

  ![anny-divide](https://user-images.githubusercontent.com/36542986/153383875-1106592f-eb48-43b4-8181-bcfb518eabfa.jpg)

Reply:
Thanks for the review, changes implemented.